shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -854,24 +782,25 @@ - (BOOL)createShell:(NSString*)entrypoint
 
   FlutterView.forceSoftwareRendering = settings.enable_software_rendering;
 
-  auto platformData = [_dartProject.get() defaultPlatformData];
+  auto platformData = [self.dartProject defaultPlatformData];
 
   SetEntryPoint(&settings, entrypoint, libraryURI);
 
-  NSString* threadLabel = [FlutterEngine generateThreadLabel:_labelPrefix];
+  NSString* threadLabel = [FlutterEngine generateThreadLabel:self.labelPrefix];
   _threadHost = std::make_shared<flutter::ThreadHost>();
   *_threadHost = MakeThreadHost(threadLabel, settings);
 
-  // Lambda captures by pointers to ObjC objects are fine here because the
-  // create call is synchronous.
+  __weak FlutterEngine* weakSelf = self;
   flutter::Shell::CreateCallback<flutter::PlatformView> on_create_platform_view =
-      [self](flutter::Shell& shell) {
-        [self recreatePlatformViewController];
-        self->_platformViewsController->SetTaskRunner(
+      [weakSelf](flutter::Shell& shell) {
+        FlutterEngine* strongSelf = weakSelf;
```

Comment:
This code isn't safe if the comment you removed was wrong (and even if it happens to currently be safe, we shouldn't leave it this way) because the block is unconditionally using `strongSelf`. There should be a:

```objc
if (!strongSelf) {
  return;
}
```

right after this line.

Reply:
Nice catch -- did this in a couple others but totally missed it here. This block is expected to return a `std::unique_ptr<flutter::PlatformView>`, so returning a null one here in the hopes that whoever calls this is careful enough to check it (they are, at the moment):
https://github.com/flutter/engine/blob/e03449ed1ee1f18b06adb8c250dc0781be66db67/shell/common/shell.cc#L236-L239